packages/web3-eth/src/errors.ts

```diff
@@ -1,21 +1,159 @@
 /* eslint-disable max-classes-per-file */
 
-import { Web3Error } from 'web3-utils';
+import { Numbers, Web3Error } from 'web3-utils';
 
 export class InvalidTransactionWithSender extends Web3Error {
-	// eslint-disable-next-line @typescript-eslint/no-explicit-any
-	public constructor(value: any) {
-		// TODO Discuss this naive approach to logging object
-		// Does not account for non JSON properties
+	public constructor(value: unknown) {
 		super(JSON.stringify(value), 'invalid transaction with sender');
 	}
 }
 
 export class InvalidTransactionCall extends Web3Error {
-	// eslint-disable-next-line @typescript-eslint/no-explicit-any
-	public constructor(value: any) {
-		// TODO Discuss this naive approach to logging object
-		// Does not account for non JSON properties
+	public constructor(value: unknown) {
 		super(JSON.stringify(value), 'invalid transaction call');
 	}
 }
+
+export class MissingCustomChainError extends Web3Error {
+	public constructor() {
+		// TODO - Discuss what to pass as value
+		super('N/A', 'If tx.common is provided it must have tx.common.customChain');
+	}
+}
+
+export class MissingCustomChainIdError extends Web3Error {
+	public constructor() {
+		super(
+			// TODO - Discuss what to pass as value
+			'N/A',
+			'If tx.common is provided it must have tx.common.customChain and tx.common.customChain.chainId',
+		);
+	}
+}
+
+export class ChainIdMismatchError extends Web3Error {
+	public constructor(value: { txChainId: unknown; customChainId: unknown }) {
+		super(
+			JSON.stringify(value),
+			'Chain Id doesnt match in tx.chainId tx.common.customChain.chainId',
+		);
+	}
+}
+
+export class CommonOrChainAndHardforkError extends Web3Error {
+	public constructor() {
+		// TODO - Discuss what to pass as value, would it make sense to pass common, chain, and hardfork for error message here?
+		super(
+			'N/A',
+			'Please provide the @ethereumjs/common object or the chain and hardfork property but not all together.',
+		);
```

Comment:
`Web3Error` requires a string value to be passed to be included in the error message. For this error it doesn't make sense to pass a value in and I wasn't sure what to pass to `Web3Error`'s constructor